vault/core.go

```diff
@@ -1123,7 +1124,7 @@ func (c *Core) unsealPart(ctx context.Context, seal Seal, key []byte, useRecover
 		// keys setup, nor 2) seals that support recovery keys but not stored keys.
 		// If insufficient shares are provided, shamir.Combine will error, and if
 		// no stored keys are found it will return masterKey as nil.
-		if seal.StoredKeysSupported() == StoredKeysSupportedGeneric {
+		if seal.StoredKeysSupported() != StoredKeysNotSupported {
```

Comment:
No change needed, just curious if it's possible for this to be anything other than `StoredKeysSupportedGeneric` since we only get into this block if the seal supports recovery keys?

Reply:
Ah, you're right, this is unnecessary.  It was a first attempt to fix the problem.  Will revert.